luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/LuthierIndustrialPark.java

```diff
@@ -67,10 +72,35 @@ Dimension getDimension(String dimensionName) {
         return dimensionDictionary.findByApiName(dimensionName);
     }
 
-/*
-    SearchProvider getSearchProvider(String searchProviderName);
-    KeyValueStore getKeyValueStore(String keyValueStoreName);
-*/
+    /**
+     * Bare minimum that can work
+     */
+
+    // TODO: Magic values!
+    int MAGIC_queryWeightLimit = 10000;
+    String MAGIC_luceneIndexPath = "path";
+    int MAGIC_maxResults = 10000;
+    public SearchProvider getSearchProvider(String searchProviderName) {
+        switch (searchProviderName) {
+            case "com.yahoo.bard.webservice.data.dimension.impl.NoOpSearchProvider":
+                return new NoOpSearchProvider(MAGIC_queryWeightLimit);
+            case "com.yahoo.bard.webservice.data.dimension.impl.LuceneSearchProvider":
+                return new LuceneSearchProvider(MAGIC_luceneIndexPath, MAGIC_maxResults);
+            default:
+                return new ScanSearchProvider();
```

Comment:
When you build a search provider lua, 
each of these constructor statements should be wrapped by a SearchProviderFactory.
The map of names to search provider factories should be passed into the LuthierIndustrialPark via constructor.

Reply:
Thanks for the comment, these changes are refactored in another branch in my repo, and I have not fully passed the regression test to feel comfortable about a new PR. Can we ignore these for now and I will rebase and open another PR later? 
The branch is here: 
https://github.com/gab-umich/fili/commits/searchProviderTemplate